src/components/Header/Header.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  FormEvent,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import classNames from 'classnames';
+
+import './Header.scss';
+import { TodosContext } from '../TodosContext';
+import { createTodo, updateTodo } from '../../api/todos';
+
+const USER_ID = 11677;
+
+export const Header: React.FC = () => {
+  const {
+    todos,
+    dispatch,
+    setErrorMessage,
+    tempTodo,
+    setTempTodo,
+    setToggledIds,
+  } = useContext(TodosContext);
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isToggleAllActive, setIsToggleAllActive] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const findIsAllActive = () => {
+    return !todos.some(todo => todo.completed === false);
+  };
+
+  const isAllActive = findIsAllActive();
+
+  useEffect(() => {
+    setIsToggleAllActive(isAllActive);
+  }, [isAllActive]);
```

Comment:
`isAllActive` is a common variable and recreates after any state, context value, change, so this useEffect works on any state change
use `useMemo` hook to store it (write there logic of `findIsAllActive` function)